Include/cpython/ceval.h

```diff
@@ -33,3 +33,5 @@ _PyEval_RequestCodeExtraIndex(freefunc f) {
 
 PyAPI_FUNC(int) _PyEval_SliceIndex(PyObject *, Py_ssize_t *);
 PyAPI_FUNC(int) _PyEval_SliceIndexNotNone(PyObject *, Py_ssize_t *);
+
+PyAPI_FUNC(PyObject *) PyEval_GetFrameLocals(void);
```

Comment:
At least in 3.12, should we name this with a leading underscore and keep it in the internal API? It's a bit late to add a public API function, and it doesn't seem necessary to fix the bug.

Or do you think this is needed for C extensions that call `PyEval_GetLocals()`?

Reply:
If there are C extensions calling `PyEval_GetLocals()` and running into the equivalent of the OP case, then they would want to switch to `PyEval_GetFrameLocals()` to fix that. I'm skeptical that any such cases will exist, but that's why I left it public. Open to whatever @Yhg1s prefers here.